main.go

```diff
@@ -15,19 +15,15 @@ func (b bin) String() string {
 	return fmt.Sprintf("%b", b)
 }
 
-func stringInSlice(a string, list []string) bool {
-	for _, b := range list {
-		if b == a {
-			return true
-		}
-	}
-	return false
-}
-
 func main() {
 	NUMBER_RESTARTS := 100
 
-	waitingReasons := []string{"CrashLoopBackOff", "ErrImagePull", "Completed", "Failed"}
+	waitingReasons := map[string]struct{}{
```

Comment:
@timjonesdev Here I'm trying out the map and then below the check for the key within the map. Does this look about right?